File: parking-lot/ParkingTicket.cpp
#include <string>
#include "Payment.cpp"

using namespace std;

class ParkingTicket 
{
private:
	int ticketNo;
	string timestamp;
	string exit;
	double amount;
	Payment payment;

public:
	ParkingTicket(
		int ticketNo,
		const string& timestamp,
		const string& exit,
		double amount,
		Payment payment
	) : ticketNo(ticketNo), timestamp(timestamp), exit(exit), amount(amount), payment(payment) 
	{}
};

File: parking-lot/core/domain/entities/parking-ticket.ts
import { Vehicle } from "./vehicle/vehicle";
import { Payment } from "./payment/payment";
import { Entrance } from "./entrance";
import { Exit } from "./exit";

export class ParkingTicket {
    public constructor(
        private readonly ticketNo: number,
        private readonly timestamp: Date,
        private readonly exit: Date,
        private readonly amount: number,
        private readonly status: boolean,
    
        private readonly vehicle: Vehicle,
        private readonly payment: Payment,
        private readonly entrance: Entrance,
        private readonly exitIns: Exit,        
    ) {}
}